### ISSUE TITLE
Publish our new package template as a standalone tool that everyone can use

### ISSUE THREAD

**Issue by KyleAMathews** [gatsbyjs/gatsby#3900]
To add a new package to this repo, you can run `npm run plop` and then choose `package` and then it sets up a new package for you — see https://github.com/gatsbyjs/gatsby/tree/master/plop-templates/package

It'd be cool to have a standalone tool for others to use when creating a new plugin to distribute as it's not always obvious how to create new NPM packages and publish them.

This along with a short tutorial on publishing a new NPM package would be 🔥

**Comment by shannonbux** [comment]
This is the #3 most wanted thing in Gatsby tutorials. A tutorial on how to create and/or modify an existing plugin. Making it easy to create and publish would be top notch.

**Comment by m-allanson** [comment]
Maybe this could be rolled into `gatsby-cli`?

Being able to do `gatsby plugin my-new-plugin` could be nice.

**Comment by KyleAMathews** [comment]
@m-allanson that makes sense!

**Comment by shannonbux** [comment]
@KyleAMathews I'm thinking of adding this to docs projects. Does that make sense?

**Comment by KyleAMathews** [comment]
@shannonbux 👍

**Comment by shannonbux** [comment]
Summary of what interviewees said about needing help building their own packages/plugins: https://slack-files.com/T6VN36NMP-F8RP27E4A-928a785a43

**Comment by Jaikant** [comment]
@m-allanson I prefer npm install as it gives me the native feel of react and npm. Enhancing the cli though could help those not familiar to npm

**Comment by Redmega** [comment]
All good points @shannonbux 

For the record, I strongly identify with:
> People coming from React might need to know when they need a plugin vs. when can they do something natively

**Comment by shannonbux** [comment]
@benjaminhoffman have you seen this?

**Comment by benjaminhoffman** [comment]
love this.  thx for the ping.  going thru the steps to create a plugin now.  hopefully have it wrapped up in a week or so and will write a 'how to post'.

**Comment by shannonbux** [comment]
Will this be closed once @jlengstorf plugin work is done (the tutorial, the sample READme template thing, etc.)?

**Comment by shannonbux** [comment]
@KyleAMathews a lot of people liked this idea, wondering when/if to bring it back into people's minds!

**Comment by KyleAMathews** [comment]
Definitely still worth doing!

**Comment by jlengstorf** [comment]
This should be rolled up into the CLI work that I still need to define.

**Comment by KyleAMathews** [comment]
Due to the high volume of issues, we're closing out older ones without recent activity. Please open a new issue if you need help!
